experiments/yarn/src/main/scala/org/apache/gearpump/experiments/yarn/master/ResourceManagerCallbackHandler.scala

```diff
@@ -1,21 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package org.apache.gearpump.experiments.yarn.master
 
+import java.util.concurrent.atomic.AtomicInteger
+
 import akka.actor.ActorRef
-import org.apache.hadoop.yarn.client.api.async.AMRMClientAsync
-import org.apache.hadoop.yarn.api.records.ContainerStatus
+import org.apache.gearpump.experiments.yarn.AppConfig
+import org.apache.gearpump.experiments.yarn.Constants._
 import org.apache.gearpump.util.LogUtil
-import org.apache.hadoop.yarn.api.records.NodeReport
-import java.util.concurrent.atomic.AtomicInteger
+import org.apache.hadoop.yarn.api.records.{Container, ContainerExitStatus, ContainerStatus, NodeReport}
+import org.apache.hadoop.yarn.client.api.async.AMRMClientAsync
 import org.slf4j.Logger
-import org.apache.hadoop.yarn.api.records.Container
-import org.apache.gearpump.experiments.yarn.Actions._
-import scala.collection.JavaConversions._
+
 import scala.collection.JavaConverters._
-import org.apache.hadoop.yarn.api.records.ContainerExitStatus
-import org.apache.gearpump.experiments.yarn.Constants._
-import org.apache.gearpump.experiments.yarn.AppConfig
 
 class ResourceManagerCallbackHandler(appConfig: AppConfig, am: ActorRef) extends AMRMClientAsync.CallbackHandler {
```

Comment:
Besides, in function onContainersAllocated, there is call
``` am ! LaunchContainers(allocatedContainers.asScala.toList)```

I think it should only send events like ContainerAllocated instead of sending command like LaunchContainers
As it is the listener's responsbility to decide how to handle event like ContainerAllocated